zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -342,6 +344,97 @@ pub fn send_single_step_proposed_transfer<T: ShieldedPoolTester>(
     );
 }
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct ConfirmationStep {
+    i: u32,
+    confirmation_requirement: u32,
+    number_of_confirmations: u32,
+    pending_balance: Zatoshis,
+    spendable_balance: Zatoshis,
+    total_balance: Zatoshis,
+}
+
+/// Tests that inputs from a source can be spent according to the default
+/// `ConfirmationPolicy`.
+///
+/// The test:
+/// - Adds funds to the wallet in a single note from an certain source.
+/// - Checks that the wallet balances are correct after N confirmations, according to
+///   the policy.
+pub fn zip_315_confirmations_test_steps<T: ShieldedPoolTester>(
+    dsf: impl DataStoreFactory,
+    cache: impl TestCache,
+    input_is_trusted: bool,
+) {
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(dsf)
+        .with_block_cache(cache)
+        .with_account_from_sapling_activation(BlockHash([0; 32]))
+        .build();
+    let account = st.test_account().cloned().unwrap();
+    let dfvk = T::test_account_fvk(&st);
+    let starting_balance = Zatoshis::const_from_u64(60_000);
+    // Add funds to the wallet in a single note, owned by the internal spending key,
+    // this is the first confirmation
+    let address_type = if input_is_trusted {
+        AddressType::Internal
+    } else {
+        AddressType::DefaultExternal
+    };
+
+    let (h, _, _) = st.generate_next_block(&dfvk, address_type, starting_balance);
+    st.scan_cached_blocks(h, 1);
+
+    // Spendable balance matches total balance at 1 confirmation.
+    assert_eq!(st.get_total_balance(account.id()), starting_balance);
+    assert_eq!(st.get_spendable_balance(account.id(), 1), starting_balance);
```

Comment:
blocking: I now am remembering that `WalletRead::get_spendable_balance` also needs to take the confirmations policy into account. Otherwise, the wallet app will not be able to use the same rules for selection of notes as it does for displaying usable balance, which will lead to bugs and/or user confusion.